src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs

```diff
@@ -117,14 +117,31 @@ private static bool IsFixableNode(SyntaxNode node)
                 };
             }
 
-            if (TryGetCommonReceiver(leftReceiver, rightReceiver, model) is var (commonReceiver, leftNames, rightNames))
+            if (TryGetCommonReceiver(leftReceiver, rightReceiver, leftTarget, rightTarget, model) is var (commonReceiver, leftNames, rightNames))
             {
                 return root =>
                 {
-                    var leftSubpattern = CreateSubpattern(leftNames, CreatePattern(leftReceiver, leftTarget, leftFlipped));
-                    var rightSubpattern = CreateSubpattern(rightNames, CreatePattern(rightReceiver, rightTarget, rightFlipped));
-                    var replacement = IsPatternExpression(commonReceiver, RecursivePattern(leftSubpattern, rightSubpattern));
-                    return root.ReplaceNode(logicalAnd, AdjustBinaryExpressionOperands(logicalAnd, replacement));
+                    // It's possible we decided to discard a pattern due to it being redundant (such as a null check
+                    // combined with a property check belonging to the same field we confirmed not being null)
```

Comment:
so i actually recommend giving examples in cases like this.  so show what sort of code the user would have started with, and what it would transform into.

Reply:
Done.